x-pack/plugins/cases/common/types/api/configure/v1.ts

```diff
@@ -69,32 +71,51 @@ export const CustomFieldsConfigurationRt = limitedArraySchema({
   fieldName: 'customFields',
 });
 
-export const TemplateConfigurationRt = rt.strict({
-  /**
-   * key of template
-   */
-  key: regexStringRt({
-    codec: limitedStringSchema({ fieldName: 'key', min: 1, max: MAX_TEMPLATE_KEY_LENGTH }),
-    pattern: '^[a-z0-9_-]+$',
-    message: `Key must be lower case, a-z, 0-9, '_', and '-' are allowed`,
-  }),
-  /**
-   * name of template
-   */
-  name: limitedStringSchema({ fieldName: 'name', min: 1, max: MAX_TEMPLATE_NAME_LENGTH }),
-  /**
-   * description of templates
-   */
-  description: limitedStringSchema({
-    fieldName: 'description',
-    min: 1,
-    max: MAX_TEMPLATE_DESCRIPTION_LENGTH,
+export const TemplateConfigurationRt = rt.intersection([
+  rt.strict({
+    /**
+     * key of template
+     */
+    key: regexStringRt({
+      codec: limitedStringSchema({ fieldName: 'key', min: 1, max: MAX_TEMPLATE_KEY_LENGTH }),
+      pattern: '^[a-z0-9_-]+$',
+      message: `Key must be lower case, a-z, 0-9, '_', and '-' are allowed`,
+    }),
+    /**
+     * name of template
+     */
+    name: limitedStringSchema({ fieldName: 'name', min: 1, max: MAX_TEMPLATE_NAME_LENGTH }),
+    /**
+     * description of templates
+     */
+    description: limitedStringSchema({
+      fieldName: 'description',
+      min: 1,
+      max: MAX_TEMPLATE_DESCRIPTION_LENGTH,
+    }),
+    /**
+     * case fields
+     */
+    caseFields: rt.union([rt.null, CaseBaseOptionalFieldsRequestRt]),
   }),
-  /**
-   * case fields
-   */
-  caseFields: rt.union([rt.null, CaseBaseOptionalFieldsRequestRt]),
-});
+  rt.exact(
+    rt.partial({
```

Comment:
super nit: Do the tags need to be `partial`? I feel like saying they're an array with minlength 0 achieves the same and is way less verbose.

Reply:
Without `partial` and with `minLength: 0` , it will still force user to add `tags:[]` param.

Reply:
yy I know, but I thought why not have `tags: []` anyway if they are the same 🤷  Anyway, not a big deal. Feel free to leave as is 👍